src/lib/TranscriptEditor/MediaPlayer/index.js

```diff
@@ -221,109 +210,90 @@ class MediaPlayer extends React.Component {
-
-      {/* Backward Btn */}
-      {this.props.mediaUrl !== null ? <button onClick={ () => { this.skipBackward() } }> {'◀◀'} </button> : ''}
-      {/* Forward Btn */}
-      {this.props.mediaUrl !== null ? <button onClick={ () => { this.skipForward() } }> {'▶▶'} </button> : ''}
-
-         ️<br/>
-      {/* Display timecodes */}
-      <code>{this.videoRef.current !== null ? secondsToTimecode(this.videoRef.current.currentTime + this.state.timecodeOffset) : '00:00:00:00'}</code>
-            /
-      <code>{this.videoRef.current !== null ? secondsToTimecode(this.videoRef.current.duration + this.state.timecodeOffset) : '00:00:00:00'}</code>
-      <br/>
-
-      <button type="button" onClick={ this.promptSetCurrentTime }>Jump To Time ⏱</button> <br/>
-
-      <button type="button" onClick={ () => { this.setTimeCodeOffset( prompt('Add a timecode offset hh:mm:ss:ff')) } }>Set Timecode Offset ⏱</button>
-      <output><code>{secondsToTimecode(this.state.timecodeOffset)}</code></output>
-      <br/>
-      {/* <label>Supported timecode formats</label>
+    let playerControlsSection;
+    if(this.props.mediaUrl !== null ){
+      playerControlsSection = <section>
+        {/* Progress bar  */}
+        <ProgressBar
+          max={ this.videoRef.current !== null ? parseInt(this.videoRef.current.duration) : 100 }
+          value={ this.videoRef.current !== null ? parseInt(this.videoRef.current.currentTime) : 0 }
+          buttonClick={ this.handleProgressBarClick.bind(this) } 
+        />
         <br/>
-        <code>hh:mm:ss:ms</code>  <code>mm:ss</code>  <code>m:ss</code> <code>m.ss</code>  <code>seconds</code>  <code>hh:mm:ss</code>
-        <br/> */}
-
-      {/* Volume Toggle */}
-      <p className={ styles.helpText }>Volume</p>
-      <label className={ styles.switch }>
-        <input type="checkbox"
-          defaultChecked="true"
-          onChange={ this.handleMuteVolume }
-          />
-        <span className={ styles.slider }></span>
-      </label>
-
-      {/* Playback Rate  */}
-      <p className={ styles.helpText }>Playback Rate
-        <b> <output >{ `x${ this.state.playBackRate }` }</output> </b>
-      </p>
-
-      <input
-        type="range"
-        min="0.2"
-        value={ this.state.playBackRate }
-        max="3.5"
-        step="0.1"
-        // list="tickmarks"
-        onChange={ this.handlePlayBackRateChange }
-        ref={ this.playbackRateInputRange }
+        {/* ▶ ❚❚ ◀◀ ▶▶ Btn - display time, set timecode offset  */}
+        <PlayerControls
+          playMedia={ this.playMedia.bind(this) }
+          isPlaying={ this.isPlaying.bind(this) }
+          skipBackward={ this.skipBackward.bind(this) }
+          skipForward={ this.skipForward.bind(this) }
+          currentTime={ this.videoRef.current !== null ? secondsToTimecode(this.videoRef.current.currentTime + this.state.timecodeOffset) : '00:00:00:00' }
```

Comment:
imo this is nicer/more readable if calculated above

Reply:
ok, wrapping currentTime and duration in getter